src/auth/dto/create_user.dto.ts

```diff
@@ -22,8 +22,8 @@ export class CreateUserDto {
   password: string;
```

Comment:
I know is not part of this PR, but I was taking a look to the regular expression you use to validate passwords.
`@Matches(/((?=.*\d)|(?=.*\W+))(?![.\n])(?=.*[A-Z])(?=.*[a-z]).*$/` and that kind of complicated regular expressions are hard to read and to maintain and can easily cause hard-to-find bugs. Here, is better to add a function to make that validation, it would be easy to understand and change and you can use it inside your `dto` with a decorator, something like `@IsPasswordValid()`